Console.Phonebook.CodeReview

samggannon opened the pull request.

ManuelE-Osorio commented on Phonebook.samggannon/Phonebook.samggannon/Services/ContactsService.cs:
```diff
@@ -0,0 +1,175 @@
+
+        while(!isValidPhoneNumber)
+        {
+            AnsiConsole.WriteLine("Invalid phone number. Please provide country and area code.");
+            AnsiConsole.WriteLine("Phone number must be at least 10 digits");
+            AnsiConsole.WriteLine("");
```
The user feedback is not entirely correct. The phone number must be 10 digits for it to be accepted.

ManuelE-Osorio reviewed (changes requested):
Hello @samggannon great work! I have been reviewing your code and you have most of the functionality already implemented. Whoever I think you are struggling right now with input validation. Can you please check the requested changes for the validation in the code and update your project? I've also left some general comments for guidance in your future projects. By the way, what do you think about EntityFramework?

samggannon commented:
Hello, thank you for the review and all the hard work the team puts in reviewing code for the C# Academy.

- Fixed the typo in the `isPhoneNumberValid` method. Now the user should be able to input a string, if it's null or empty, it will return false and prompt the user to add a phone number.
  - If it's not empty, it checks that all characters are valid digits and prompts the user accordingly if they aren't.
- Resolved the issue of passing a null or empty email string to the sender. Refactored the code to send emails after adding a contact and updating a contact.
- Resolved the "USERPROFILE" environment method to make the program more robust cross-platform.
- Refactored the Update and Delete methods. If invalid inputs are entered, it notifies the user and then it forces the user back to the main menu.
- Streamlined the email address validation by using the `System.Net.Mail` class. I let the `MailAddress` constructor parse the email address format.

>By the way, what do you think about EntityFramework?

This was my first exposure to Entity Framework. Working with basic CRUD operations are a joy in EF. I have about 2 years of experience with data entry with IBM's DB2 database syntax. I prefer it over MS SQL, particularly with handling table creations. Letting EF abstract away the table creation was so refreshing. I can't wait to delve a little deeper into Entity Framework.

ManuelE-Osorio commented:
@samggannon great work, I will approve your project! As I final suggestion, if you will work with phone numbers in the future maybe you'd like this library https://github.com/twcclegg/libphonenumber-csharp
